transmart-core-db/grails-app/services/org/transmartproject/db/userqueries/UserQueryService.groovy

```diff
@@ -1,46 +1,192 @@
 package org.transmartproject.db.userqueries
 
 import grails.transaction.Transactional
+import grails.util.Holders
+import groovy.transform.CompileStatic
+import org.hibernate.SessionFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.binding.BindingHelper
+import org.transmartproject.core.concept.ConceptsResource
 import org.transmartproject.core.exceptions.AccessDeniedException
 import org.transmartproject.core.exceptions.InvalidArgumentsException
 import org.transmartproject.core.exceptions.NoSuchResourceException
+import org.transmartproject.core.exceptions.ServiceNotAvailableException
+import org.transmartproject.core.multidimquery.query.Constraint
+import org.transmartproject.core.multidimquery.query.ConstraintFactory
+import org.transmartproject.core.userquery.SubscriptionFrequency
 import org.transmartproject.core.userquery.UserQuery
+import org.transmartproject.core.userquery.UserQueryRepresentation
 import org.transmartproject.core.userquery.UserQueryResource
+import org.transmartproject.core.userquery.UserQuerySetResource
+import org.transmartproject.core.users.AuthorisationChecks
+import org.transmartproject.core.users.LegacyAuthorisationChecks
+import org.transmartproject.core.users.PatientDataAccessLevel
 import org.transmartproject.core.users.User
+import org.transmartproject.db.clinical.ConstraintAccessChecker
 import org.transmartproject.db.querytool.Query
 
+import java.util.stream.Collectors
+
 @Transactional
+@CompileStatic
 class UserQueryService implements UserQueryResource {
 
-    @Override
-    List<UserQuery> list(User currentUser) {
-        Query.findAllByUsernameAndDeleted(currentUser.username, false)
+    @Autowired
+    UserQuerySetResource userQuerySetResource
+
+    @Autowired
+    AuthorisationChecks authorisationChecks
+
+    @Autowired
+    LegacyAuthorisationChecks legacyAuthorisationChecks
+
+    @Autowired
+    ConceptsResource conceptsResource
+
+    @Autowired
+    SessionFactory sessionFactory
+
+
+    private static void validateSubscriptionEnabled(Boolean subscribed, SubscriptionFrequency subscriptionFreq) {
```

Comment:
`Boolean subscribed` - ah ok, nice, it fixes the current issue for `subscribed == null`